### ISSUE TITLE
[contract] Verify Auth

### ISSUE THREAD

**Issue by scuellar** [GaloisInc/formal-verso#2]
The example contract for Auth is [described in the Soroban tutorial](https://developers.stellar.org/docs/smart-contracts/example-contracts/auth) and the code can be found [here](https://github.com/stellar/soroban-examples/tree/v20.0.0/auth).

**Comment by rperoutka** [comment]
I added the start of a verification script for the Auth contract in [auth.saw](https://github.com/GaloisInc/formal-verso/commit/4fe700f86a2891306909ee0a2c4e1ff8b5beb7f5). I had to add bits for the `AddressObject` type to [soroban.saw](https://github.com/GaloisInc/formal-verso/commit/0a457661a30f205cdff4a21c1c8bba6021d40c45) to enable creating a fresh variable of type `Address` for the spec. Verification currently fails when symbolic execution attempts a call to `Address::requireAuth`.
